src/lib/Embed.hs

```diff
@@ -328,6 +329,13 @@ appTryReduce f x = case f of
 ptrOffset :: MonadEmbed m => Atom -> Atom -> m Atom
 ptrOffset x i = emitOp $ PtrOffset x i
 
+unsafePtrLoad :: MonadEmbed m => Atom -> m Atom
+unsafePtrLoad x = emit $ Hof $ RunIO $ Lam $ Abs (Ignore UnitTy) $
+  (PlainArrow justIOEff, Block Empty (Op (PtrLoad x)))
+
+justIOEff :: EffectRow
+justIOEff = EffectRow [(State, theWorld)] Nothing
```

Comment:
you're not exporting this nor using it anywhere else. Make it part of a `where` block of the above def?